acl/errors.go

```diff
@@ -126,3 +128,24 @@ func PermissionDeniedByACLUnnamed(_ Authorizer, context *AuthorizerContext, reso
 	desc := NewResourceDescriptor("", context)
 	return PermissionDeniedError{Accessor: "", Resource: resource, AccessLevel: accessLevel, ResourceID: desc}
 }
+
+func RequirePermissionDeniedError(t testing.TB, err error, _ Authorizer, _ *AuthorizerContext, resource Resource, accessLevel AccessLevel, resourceID string) {
+	t.Helper()
+	if err == nil {
+		t.Fatal("An error is expected but got nil.")
+	}
+	if v, ok := err.(PermissionDeniedError); ok {
+		require.Equal(t, v.Resource, resource)
+		require.Equal(t, v.AccessLevel, accessLevel)
+		require.Equal(t, v.ResourceID.Name, resourceID)
+	} else {
+		t.Fatalf("Expected a permission denied error got %v", err)
```

Comment:
Sometimes if there's a type assertion involved, I like to get `%T` into the error messages too, so you know what the type of the value was as well.

Reply:
Good idea.